examples/sdbus_async/gatt_props.py

```diff
@@ -0,0 +1,192 @@
+            print(f"read value: {str(value)}")
+
+
+async def print_service_props(dbus: SdBus, service_name: str) -> None:
+    path = DEV_PATH + '/' + service_name
+    print('=============================================')
+    print(f"service: {service_name}")
+    service = GattServiceInterfaceAsync()
+    service._connect('org.bluez', path, bus=dbus)
+    service_properties = {
+        'UUID': service.uuid,
+        'device': service.device_path,
+        'primary': service.primary,
+        'includes paths': service.includes_paths,
+        'handle': service.handle,
+    }
+    await print_props(service_properties, "")
+
+    chars = await find_chars(dbus, path)
+    if chars:
+        for char in chars:
+            print(f" {char}")
+            await print_char_props(dbus, service_name, char)
+    return None
+
+
+async def find_chars(dbus: SdBus, path: str) -> list[str]:
+    # do D-Bus introspect to the service path and get characteristic paths
+    adapter_introspect = DbusIntrospectableAsync()
+    adapter_introspect._connect('org.bluez', path, bus=dbus)
+    s = await adapter_introspect.dbus_introspect()
+    parser = ET.fromstring(s)
+    nodes = parser.findall("./node")
+    if not nodes:
+        print("characteristic not found")
+        return []
+
+    chars = []
+    for node in nodes:
+        chars.append(node.attrib['name'])
+    return chars
+
+
+async def find_service(dbus: SdBus) -> list[str]:
+    # do D-Bus introspect to the device path and get service paths under it
+    adapter_introspect = DbusIntrospectableAsync()
+    adapter_introspect._connect('org.bluez', DEV_PATH, bus=dbus)
+    s = await adapter_introspect.dbus_introspect()
+    parser = ET.fromstring(s)
+    nodes = parser.findall("./node")
+    if not nodes:
+        print("service not found")
+        return []
+
+    services = []
+    for node in nodes:
+        print(f" {node.attrib['name']}")
+        services.append(node.attrib['name'])
+    return services
+
+
+async def main() -> None:
+
+    # connect to D-Bus
+    dbus = sd_bus_open_system()
+    device = DeviceInterfaceAsync()
+    device._connect('org.bluez', DEV_PATH, bus=dbus)
+
+    # discover the specified BLE device
+    await discover(dbus)
+
+    # connect to the device
+    print(f"Conneting to {DEV_PATH}...")
+    try:
+        await device.connect()
+    except NotImplementedError as e:
+        print(e)
+        return
+    except Exception as e:
+        print(e)
+        print(f"failed to connect: {e}")
+        return
+    print("Conneted")
+
+    # Get services and characteristics and print their properties
+    await sleep(2)
+    for i in range(3):
+        print("Find services...")
+        services = await find_service(dbus)
+        if services:
+            for service in services:
+                await print_service_props(dbus, service)
+            break
+            await sleep(1)
+
+    # Clean up
+    await device.disconnect()
+
+run(main())
```

Comment:
Do you mind putting this function call under `if __name__ == "__main__":`. It is considered a good practice. (also the `if len(sys.argv) < 2:`)